packages/medusa-react/src/hooks/admin/publishable-api-keys/queries.ts

```diff
@@ -52,3 +54,21 @@ export const useAdminPublishableApiKeys = (
   )
   return { ...data, ...rest } as const
 }
+
+export const useAdminPublishableApiKeySalesChannels = (
+  id: string,
+  query?: GetPublishableApiKeySalesChannelsParams,
+  options?: UseQueryOptionsWrapper<
+    Response<AdminSalesChannelsListRes>,
+    Error,
+    ReturnType<PublishableApiKeyQueryKeys["detail"]>
+  >
+) => {
+  const { client } = useMedusa()
+  const { data, ...rest } = useQuery(
+    adminPublishableApiKeysKeys.detail(id),
```

Comment:
We might run in to some conflicts with the caching key used here if the publishablekey is fetched using the `useAdminPublishableApiKey` hook before trying to fetch the sales channels at this point. 

To solve it I think we should use a different key for this as we also do with listing products by price list (as well as some other places). wdyt?